tbtadm/controller.cpp

```diff
@@ -653,6 +655,188 @@ catch (...)
     m_err << "Unknown exception\n";
 }
 
+void tbtadm::Controller::addToBootACL(const fs::path& dir, std::string uuid)
+{
+    auto fname = dir.filename().string();
+    auto num = fname.substr(0, fname.find('-'));
+    auto boot_acl_path = sysfsDevicesPath / (domain + num) / boot_acl;
+
+    if (!fs::exists(boot_acl_path))
+    {
+        /* Cannot find boot_acl */
+        return;
+    }
+
+    /* Read boot_acl */
+    std::stringstream stream(readAndTrim(boot_acl_path));
+    /* Check maximum allowed boot_acl string size */
+    if (stream.str().size() > 36 * 16 + 1 * 15)
+    {
+        m_out << "boot_acl contains too much characters\n";
+        return;
+    }
+
+    std::queue<std::string> acl_queue;
+
+    while(stream.good())
+    {
+        std::string substring;
+
+        getline(stream, substring, ',');
+	/* this moves also uuid to the end of the queue */
+        if (substring.size() == 36 && substring != uuid)
+        {
+            acl_queue.push(substring);
+        }
+    }
+
+    if (acl_queue.size() > 16)
+    {
+        m_out << "Too much elements in the boot_acl list\n";
+        return;
+    }
+
+    /* Add UUID to boot_acl queue */
+    if (acl_queue.size() < 16)
```

Comment:
You can test for the case that the list needs to be trimmed and `pop`, and then the `push` can happen outsides of the `if` as it happens anyway

Reply:
OK